lib/publiccloud/utils.pm

```diff
@@ -210,4 +211,29 @@ sub define_secret_variable {
     script_run("set +a");
 }
 
+# Get credentials from the Public Cloud micro service, which requires user
+# and password. The resulting json will be stored in a file.
+sub get_credentials {
+    my $url = get_var('PUBLIC_CLOUD_CREDENTIALS_URL');
```

Comment:
I'd argue that if we are calling this routine, then PUBLIC_CLOUD_CREDENTIALS_URL is required

```suggestion
    my $url = get_required_var('PUBLIC_CLOUD_CREDENTIALS_URL');
```

Reply:
makes sense, this method is to be called only with no-vault jobs.